File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;

string s, ret; 
int cnt[200], flag; 
char mid;

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cin >> s;
	if (s.length() == 1) {
	
		cout << s;	
		return 0;
	}
	
	for (char i : s) cnt[i] ++;
	
	for (int i = 'Z'; i >= 'A'; i--) {
		if (cnt[i]){
			if (cnt[i] & 1){
				flag++;
				mid = char(i);
				cnt[i]--;
			}
			if(flag == 2){
				cout << "I'm Sorry Hansoo" << '\n';
				return 0;
			}
		
			for(int j = 0; j<cnt[i]; j += 2){
				ret = char(i) + ret;
				ret += char(i);
			}
		}
	}
	
	if(mid)ret.insert(ret.begin() + ret.size() / 2, mid);
	
	cout << ret << "\n";

    return 0;
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String temp = br.readLine();

        int[] arr = new int[26];
        for (int i = 0; i < temp.length(); i++) {
            arr[temp.charAt(i) - 'A']++;
        }
        char center = ' ';
        int cnt = 0;
        for (int i = 0; i < 26; i++) {
            if (arr[i] % 2 != 0) {
                center = (char) (i + 'A');
                cnt++;
            }
        }
        String ret = "";
        for (int i = 0; i < 26; i++) {

            for (int j = 0; j < arr[i] / 2; j++) {
                ret += (char) (i + 'A');
            }
        }
        String rev = "";
        for (int i = ret.length() - 1; i >= 0; i--) {
            rev += ret.charAt(i);
        }
        if (cnt > 1){
            System.out.println("I'm Sorry Hansoo");
        }else if (center == ' '){
            System.out.println(ret+rev);
        }else{
            System.out.println(ret+center+rev);
        }

    }
}